From: KY Support <support@ky.com>
Subject: [PATCH] riscv: add KY X1 TCM CSR register support

Add CSR_TCMCFG and TCM_EN definitions for KY X1 SoC TCM (Tightly Coupled Memory)
access control.

Signed-off-by: KY Support <support@ky.com>
---
 arch/riscv/include/asm/csr.h | 6 ++++++
 1 file changed, 6 insertions(+)

--- a/arch/riscv/include/asm/csr.h
+++ b/arch/riscv/include/asm/csr.h
@@ -393,6 +393,12 @@
 #define CSR_VTYPE		0xc21
 #define CSR_VLENB		0xc22

+#ifdef CONFIG_SOC_KY_X1
+/* TCM enable register */
+#define CSR_TCMCFG		0x5db
+#define TCM_EN			_AC(0x00000001, UL) /* TCM Access Enable */
+#endif
+
 #ifdef CONFIG_RISCV_M_MODE
 # define CSR_STATUS	CSR_MSTATUS
 # define CSR_IE		CSR_MIE
